### ISSUE TITLE
Polymorphic U3V class implementation

### ISSUE THREAD

**Issue by iitaku** [fixstars/ion-kit#245]
U3V class has flags to control behavior.  

- simulation mode
- camera mode
- frame sync

These can be simplified and reorganized using class polymorphism.

**Comment by xinyuli1204** [comment]
after merge this https://github.com/fixstars/ion-kit/pull/242

The log becomes

```
C:\Users\xinyu\Desktop\tutorials\python>python tutorial1_display_2cam.py
[2024-02-28 05:35:45.239] [ion] [debug] ion-kit version is v1.6.0-22-g47574cd
[2024-02-28 05:35:45.251] [ion] [info] Start building pipeline
[2024-02-28 05:35:45.252] [ion] [info] Builder::register_disposer
[2024-02-28 05:35:45.252] [ion] [debug] U3V::release_instance() :: is called
[2024-02-28 05:35:46.013] [ion] [debug] Inferred arguments stub
[2024-02-28 05:35:46.014] [ion] [debug] Generating arguments instance
[2024-02-28 05:35:46.014] [ion] [debug]   #0 0x20f3d7a4468
[2024-02-28 05:35:46.015] [ion] [debug]   #1 0x20f3d7a4048
[2024-02-28 05:35:46.015] [ion] [info] Create U3V instance: 638e7550-4210-47ee-bc02-29b252780a15
```

Why  `[2024-02-28 05:35:45.252] [ion] [debug] U3V::release_instance() ::` is called after` Start building pipeline`

ion-kit 1.6.0 shows
```
C:\Users\xinyu\Desktop\tutorials\python>python tutorial1_display_2cam.py
[2024-02-28 05:40:49.309] [ion] [debug] ion-kit version is v1.6.0
[2024-02-28 05:40:49.346] [ion] [info] Start building pipeline
[2024-02-28 05:40:49.361] [ion] [info] Builder::register_disposer
[2024-02-28 05:40:50.488] [ion] [info] Create U3V instance: 4acb44ce-2679-4371-a099-ad408910ed73
[2024-02-28 05:40:50.492] [ion] [debug] U3V:: 23-11-18 : updating obtain and write
[2024-02-28 05:40:50.492] [ion] [info] Using aravis-0.8.31
[2024-02-28 05:40:53.813] [ion] [info] Creating U3V instance with 2 devices...
```

rt_u3v.h  and bb.h are all the old ones from tag v1.6.0 (old means without fake camera and  polymorphic class )

**Comment by iitaku** [comment]
Resolved in #254
